### ISSUE TITLE
Allow generated parser to implement interface/extend a class

### ISSUE THREAD

**Issue by zlikavac32** [axmachado/pacc#2]
Would it make sense for a generated parser to implement an interface or extend a class.

For some complex password logic it's useful to have an abstract class that encapsulated that logic and in .y file just call those methods?

When providing the parser to the rest of the code, it would be helpful to orient towards interfaces so that test can be conducted without concrete parser instance.

Would it be ok to add those to option like `extends = class` or `implements = for now just one intreface?`

**Comment by axmachado** [comment]
Hello, Marijan

It totally makes sense for me. Would you be able to provide a patch/pull request? If you can't do it, I will try to work it out next weekend. Too busy this week to do it in working hours.

**Comment by zlikavac32** [comment]
Ok, I'll see what I can do. It's not rush :). Should `LRParserBase` then be embedded into the generated parser? That would make it consistent so that both LR and RD can have extend and interface.

**Comment by axmachado** [comment]
I think embedding the `LRParserBase` into the generated parser is not the most readable solution. Maybe we can go for something like an `LRParserRunner` class that receives the generated parser as  parameter for it's constructor, so the main LR logic is still in a separated class and it's easier to read and maintain than the code embedded into the generated parser.

**Comment by zlikavac32** [comment]
Yes, that would be ok. Shall we go with the interface support first and then see what/how next?

**Comment by axmachado** [comment]
I think it's not so difficult to support the `implements` and `extends` together.

**Comment by zlikavac32** [comment]
@axmachado Ok, I can look into it next week
